### PR TITLE
rfc: proposed docs layout [DET-4061]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

Proposed new content is mostly around debuggability and introductory porting code.  Most of the rest of the work would be just shuffling existing docs around and touching them up.

That said, there's a lot of work here.  I'd expect the shuffle-and-touchup work alone to take at least two weeks, if we want the result to feel seamless and intentional.

### CODE REVIEW

File: layout [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            prereqs:
+                docker works
+            goals:
+                have working installation (docker method)
+                have cli installed
+                    how: Install Determined CLI
+                hands-on use of cli and webui (tensorboard?), and nothing else
+            details:
+                use one-variable model (copy/paste rather than download code)
+N     * Porting A PyTorch Model
```
Comment (neilconway) [inline layout:88]: I think we should avoid the phrase "porting". Alternatives:

"Building Your First PyTorch Model in Determined"
"Using PyTorch with Determined"

etc.

File: layout [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+                have a working installation
+            steps:
+                start with a one-variable model  (use mnist instead?)
+                show a DummyContext with a skeleton training loop
+                implement build_model + train_batch
+                implement build_*_data_loader
+                run in local test mode
+                run in remote test mode
+                run a full experiment
+            (is that too many steps?)
```
Comment (neilconway) [inline layout:99]: Doesn't seem like too many steps to me.

File: layout [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            top: Experiment Lifecycle
+        Hyperparameter Tuning
+        Notebooks
+        Tensorboard
+        Model Management
+        System Components (basically a glossary)
+            top: System Architecture
+            top: Terminology and Concepts
+  * Training Models
+        Model APIs  (what should go here, other than links to referece docs?)
```
Comment (neilconway) [inline layout:125]: I don't think we need a ton of content here, but might be worth covering

* Why we need a trial API in the first place
* Concepts that apply to all of the trial APIs (e.g., trial context)
* Concepts that apply to each specific API in turn (e.g., how/why the PyTorch API works the way it does vs. how the Keras API works).

File: layout [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+            top: Effective Distributed Training
+                top: Optimizing Distributed Training
+        Reproducibility
+            top: Reproducibility
+        Experiment Configuration
+            how: Environment Configuration
+            ref: Experiment Configuration
+            ref: Configuration Templates
+            link to how: Environment Configuration
+N       Debugging Models in Determined
```
Comment (neilconway) [inline layout:144]: Great idea

**Review by neilconway — COMMENTED** [review]
Overall, looks great to me!

File: layout (hunk 3 above)
Comment (rb-determined-ai) [reply layout:125]: Stealing this for the RFC

File: layout (hunk 1 above)
Comment (rb-determined-ai) [reply layout:88]: > "Building Your First PyTorch Model in Determined"

To me this sounds too much like "starting from scratch", I'd be afraid people who needed to port a model might not recognize the title as what they need

What about "Starting with an existing PyTorch Model"?

Or "Train an existing PyTorch model with Determined"?

Or "Bring an existing PyTorch model into Determined"?

Or "Structuring a PyTorch model for Determined"?

File: layout [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        top: System Architecture
+        top: Determined on AWS
+        top: Determined on GCP
+        top: Elastic Infrastructure
+        top: Dynamic Agents on AWS
+        top: Dynamic Agents on GCP
+        top: Terminology and Concepts
+X       top: User Interfaces
+        top: Users
+X       top: YAML Topic Guide
```
Comment (katport) [inline layout:52]: the reason we added this was for users who have not worked closely with YAMLs and didn't know the structure. If they format for example bind_mounts wrong, it displays a yaml error because we can't interpret it. If we plan on removing it, I think we should make a concious effort to make sure formatting problems are described elsewhere

File: layout [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        top: Reproducibility
+        top: Scheduling
+        top: System Architecture
+        top: Determined on AWS
+        top: Determined on GCP
+        top: Elastic Infrastructure
+        top: Dynamic Agents on AWS
+        top: Dynamic Agents on GCP
+        top: Terminology and Concepts
+X       top: User Interfaces
```
Comment (katport) [inline layout:50]: The reason we added this page was because at the time we did not reference -m anywhere and did not have any page that kind describes how our CLI is structured (outside the cli man). Just something to think about.

**Review by katport — APPROVED** [review]

File: layout (hunk 1 above)
Comment (davidhershey) [reply layout:88]: "Train an existing PyTorch model with Determined" is my favorite

**Review by davidhershey — COMMENTED** [review]
Overall new structure looks good to me -- I think we need to reduce clutter and this seems to be a good path to that.

File: layout (hunk 5 above)
Comment (rb-determined-ai) [reply layout:52]: What if we just improved the error logging?  For example, an error in unmarshaling the bind_mounts could include text on how a correct bind_mounts would look?

File: layout (hunk 6 above)
Comment (rb-determined-ai) [reply layout:50]: Good to know.  Maybe we keep this doc or canabalize it into a couple different places then.

**Comment by rb-determined-ai** [comment]
I'm going to close this as "good enough to move forward", thank you all for your feedback.
